exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -252,13 +257,48 @@ func (prwe *PrwExporter) handleSummaryMetric(tsMap map[string]*prompb.TimeSeries
 }
 
 // export sends a Snappy-compressed WriteRequest containing TimeSeries to a remote write endpoint in order
-func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.TimeSeries) error {
-	// Calls the helper function to convert the TsMap to the desired format
-	req, err := wrapTimeSeries(tsMap)
+func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.TimeSeries) []error {
+	var errs []error
+	// Calls the helper function to convert and batch the TsMap to the desired format
+	requests, err := batchTimeSeries(tsMap, maxBatchByteSize)
 	if err != nil {
-		return consumererror.Permanent(err)
+		errs = append(errs, consumererror.Permanent(err))
+		return errs
+	}
+
+	input := make(chan *prompb.WriteRequest, len(requests))
+	for _, request := range requests {
+		input <- request
+	}
+	close(input)
+
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+
+	wg.Add(maxConcurrentRequests) // used to wait for workers to be finished
+
+	// Run concurrencyLimit of workers until there
+	// is no more requests to execute in the input channel.
+	for i := 0; i < maxConcurrentRequests; i++ {
+		go func() {
```

Comment:
Nit: alternatively, these goroutines could be started once when the is Started instead of doing it every time export is called, which may add a bit of overhead if the export requests are tiny.

Reply:
Hmm, I added the limit you mentioned before (about spawning the minimum number of goroutines necessary), so I don't think we would have as much overhead as before. You're right that it would be better if we started these goroutines when the exporter itself is started. We'll create an issue for this and address it in a separate PR.

Reply:
@tigrannajaryan I wonder if it's your experience that they are adding more overhead. It's so cheap to start new goroutines (even though not cost-free) that I've seen it relatively rare in Go to reuse goroutines. It's one of the goals of the goroutines to be cheap so the devs don't care about goroutine pools, etc. We might add this as a TODO to take a look at but given how net/http spawns gorotuines each time, I don't think this is an improvement worth investigating right now.

Reply:
It is likely not going to be a measurable difference for typical batch size. Was more of a though exercise, not an ask to change anything. :-)
What you have is fine.